CMakeLists.txt

```diff
@@ -118,28 +118,23 @@ include(cmake/SetupBasics.cmake)
 include(cmake/SetupPackages.cmake)
 
 if (RAJA_ENABLE_CUDA)
-  if (DEFINED CUDA_ARCH)
-    if (CUDA_ARCH MATCHES "^sm_*")
-      if ("${CUDA_ARCH}" STRLESS "sm_35")
-        message( FATAL_ERROR "RAJA requires minimum CUDA compute architecture of sm_35")
-      endif()
-    endif()
-    if (CUDA_ARCH MATCHES "^compute_*")
-      if ("${CUDA_ARCH}" STRLESS "compute_35")
-        message( FATAL_ERROR "RAJA requires minimum CUDA compute architecture of compute_35")
-      endif()
+  if (DEFINED CMAKE_CUDA_ARCHITECTURES)
+    if ("${CMAKE_CUDA_ARCHITECTURES}" STRLESS "35")
+      message( FATAL_ERROR "RAJA requires minimum CUDA compute architecture of 35")
     endif()
   else()
-    message(STATUS "CUDA compute architecture set to RAJA default sm_35 since it was not specified")
-    set(CUDA_ARCH "sm_35" CACHE STRING "Set CUDA_ARCH to RAJA minimum supported" FORCE)
+    message(STATUS "CUDA compute architecture set to RAJA default 35 since it was not specified")
+    set(CMAKE_CUDA_ARCHITECTURES "35" CACHE STRING "Set CMAKE_CUDA_ARCHITECTURES to RAJA minimum supported" FORCE)
```

Comment:
This makes me wonder if setting CMAKE_HIP_ARCHITECTURES to 90a instead of gfx90a would work as intended.

Reply:
I tried it, but it doesn't seem to work.

Reply:
CMake internally prepends "sm_" to the CUDA compute architecture when using CMAKE_CUDA_ARCHITECTURES. I initially tried to pass sm_70 and CMake died with an error b/c 'sm_sm_70' is not a know arch.

I didn't touch the HIP arch specification because it seems to be working.

Reply:
If you set GPU_TARGETS and AMDGPU_TARGETS you can avoid building for every gpu under the sun. I'm not sure what CMAKE_HIP_ARCHITECTURES does other than quiet the error that happens if you don't set it.

Reply:
Please update the HIP build scripts in this PR branch if you want.

Reply:
Ok, I'll do that.

Reply:
Thank you. I'm paranoid so you may want to verify that it picks the right compute arch on corona as well.